File: 3184-count-pairs-that-form-a-complete-day-i/3184-count-pairs-that-form-a-complete-day-i.py
class Solution:
    def countCompleteDayPairs(self, hours: List[int]) -> int:
        n = len(hours)
        count = 0
        for i in range(n-1):
            for j in range(i+1, n):
                if (hours[i] + hours[j]) % 24 == 0:
                    count += 1
        return count

File: 3184-count-pairs-that-form-a-complete-day-i/3184-count-pairs-that-form-a-complete-day-i.cpp
class Solution {
public:
    int countCompleteDayPairs(vector<int>& hours) {
        unordered_map<int, int> mp;
        int count = 0;
        for (int hour : hours) {
            int mod = hour % 24;
            int com = (24 - mod) % 24;
            if (mp.find(com) != mp.end()) {
                count += mp[com];
            }
            mp[mod]++;
        }
        return count;
    }
};



// class Solution {
// public:
//     int countCompleteDayPairs(vector<int>& hours) {
//         int n = hours.size();
//         int count = 0;
//         for (int i = 0; i < n - 1; i++) {
//             for (int j = i + 1; j < n; j++) {
//                 if ((hours[i] + hours[j]) % 24 == 0) {
//                     count++;
//                 }
//             }
//         }
//         return count;
//     }
// };